packages/core-utils/src/coders/ecdsa-coder.ts

```diff
@@ -194,10 +194,39 @@ class Eip155TxCoder extends DefaultEcdsaTxCoder {
  * ctcCoder  *
  ************/
 
+function encode(data) {
+  // TODO(annieke): establish API for how to decide which type
+  const type = 1 // parseInt(data.slice(0, 2), 16)
```

Comment:
@tynes the data transport layer only generic decodes which you've written for already - for encode, what's the preferred way to established which type we should encode to? or is there a way to tell from the decoded tx?

Reply:
I think we need to add a type field to this object:

```
      const eip155TxData = {
        sig: {
          v: 1,
          r: '0x' + '11'.repeat(32),
          s: '0x' + '22'.repeat(32),
        },
        gasLimit: 500,
        gasPrice: 100,
        nonce: 100,
        target: '0x' + '12'.repeat(20),
        data: '0x' + '99'.repeat(10),
      }
```

So that this `encode` function could look at the type and pass through to the appropriate coder. We may need to update a typescript type so that typescript builds

Reply:
will do!